output/typescript/types.ts

```diff
@@ -3977,7 +3977,7 @@ export interface ShardRecovery {
 }
 
 export interface IndexSegment {
-  shards: Record<string, ShardsSegment>
+  shards: Record<string, ShardsSegment | Array<ShardsSegment>>
```

Comment:
Can this ever be just `ShardsSegment` and not `Array<ShardsSegment>` ? 

I think we might also want to introduce a `@cleanup("reason")` reason annotation @delvedor @SylvainJuge to start annotating server responses that we think should be more regular?

Reply:
I think you pinged the Sylvain on that comment 😄